executor/executor.go

```diff
@@ -122,35 +129,64 @@ func (e *Executor) Run(conflicts state.Keys, f func() error) {
 	e.tasks[id] = t
 
 	// Record dependencies
-	for k := range conflicts {
+	for k, v := range conflicts {
 		latest, ok := e.edges[k]
 		if ok {
-			lt := e.tasks[latest]
+			lt := e.tasks[latest.TaskID]
 			if !lt.executed {
 				if t.dependencies == nil {
 					t.dependencies = set.NewSet[int](defaultSetSize)
 				}
-				t.dependencies.Add(lt.id)
 				if lt.blocking == nil {
 					lt.blocking = set.NewSet[int](defaultSetSize)
 				}
-				lt.blocking.Add(id)
+				// key has ONLY a Read permission
+				if v == state.Read {
+					// lt contains either Allocate or Write permission. If lt had only a Read
+					// permission, no dependency or blocking would be added
+					if latest.Permissions.Has(state.Allocate) || latest.Permissions.Has(state.Write) {
+						t.dependencies.Add(lt.id) // t depends on lt to execute
+						lt.blocking.Add(id)       // lt is blocking this current task
```

Comment:
You need to register the type of blocker this is `state.Read` so that when complete, `lt` can potentially unblock multiple ids.